File: 205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length() != t.length())
        return false;

        HashMap<Character,Character> map1 = new HashMap<>();

        for(int i=0;i<s.length();i++){
            if(map1.containsKey(s.charAt(i))){
                if(map1.get(s.charAt(i)) != t.charAt(i))
                return false;
            }
            else{
                map1.put(s.charAt(i),t.charAt(i));
            }
        }

        map1.clear();

        for(int i=0;i<s.length();i++){
            if(map1.containsKey(t.charAt(i))){
                if(map1.get(t.charAt(i)) != s.charAt(i))
                return false;
            }
            else{
                map1.put(t.charAt(i),s.charAt(i));
            }
        }

        return true;



    }
}

File: 205-isomorphic-strings/isomorphic-strings.c
bool isIsomorphic(char* s, char* t) {
    int map1[256] = {0};
    int map2[256] = {0};

    for(int i=0;i<strlen(s);i++){
        
        if(map1[s[i]] == 0){
            map1[s[i]] = t[i];
        }
        else{
            if(map1[s[i]] != t[i])
            return false;
        }
    }


    for(int i=0;i<strlen(s);i++){
        if(map2[t[i]] == 0){
            map2[t[i]] = s[i];
        }
        else{
            if(map2[t[i]] != s[i])
            return 0;
        }
    }

    return 1;
}